src/Symfony/Component/Security/Http/AccessToken/Oidc/OidcTokenHandler.php

```diff
@@ -92,44 +99,51 @@ public function getUserBadgeFrom(string $accessToken): UserBadge
             throw new \LogicException('You cannot use the "oidc" token handler since "web-token/jwt-signature" and "web-token/jwt-checker" are not installed. Try running "composer require web-token/jwt-signature web-token/jwt-checker".');
         }
 
-        if (!$this->discoveryCache && !$this->signatureKeyset) {
+        if (!$this->discovery && !$this->signatureKeyset) {
             throw new \LogicException('You cannot use the "oidc" token handler without JWKSet nor "discovery". Please configure JWKSet in the constructor, or call "enableDiscovery" method.');
         }
 
         $jwkset = $this->signatureKeyset;
-        if ($this->discoveryCache) {
-            try {
-                $oidcConfiguration = json_decode($this->discoveryCache->get($this->oidcConfigurationCacheKey, function (): string {
-                    $response = $this->discoveryClient->request('GET', '.well-known/openid-configuration');
-
-                    return $response->getContent();
-                }), true, 512, \JSON_THROW_ON_ERROR);
-            } catch (\Throwable $e) {
-                $this->logger?->error('An error occurred while requesting OIDC configuration.', [
-                    'error' => $e->getMessage(),
-                    'trace' => $e->getTraceAsString(),
-                ]);
-
-                throw new BadCredentialsException('Invalid credentials.', $e->getCode(), $e);
-            }
-
-            try {
-                $jwkset = JWKSet::createFromJson(
-                    $this->discoveryCache->get($this->oidcJWKSetCacheKey, function () use ($oidcConfiguration): string {
-                        $response = $this->discoveryClient->request('GET', $oidcConfiguration['jwks_uri']);
+        if ($this->discovery) {
+            $jwkset = new JWKSet([]);
+            foreach ($this->discovery as ['cache' => $discoveryCache, 'client' => $discoveryClient, 'oidcConfigurationCacheKey' => $oidcConfigurationCacheKey, 'oidcJWKSetCacheKey' => $oidcJWKSetCacheKey]) {
```

Comment:
I'd suggest removing the keys:
```suggestion
            foreach ($this->discovery as [$discoveryCache, $discoveryClient, $oidcConfigurationCacheKey, $oidcJWKSetCacheKey]) {
```

Reply:
That doesn't work, since it won't match the array shape:
```php
    /**
     * @var list<array{
     *     cache: CacheInterface,
     *     client: HttpClientInterface,
     *     oidcConfigurationCacheKey: string,
     *     oidcJWKSetCacheKey: string
     * }>
     */
    private array $discovery = [];
```

Unless you want me to remove this type information?

Reply:
It's internal, so I removed the keys.